Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -590,8 +590,11 @@ WarpXParticleContainer::GetChargeDensity (int lev, bool local)
     const auto& ba = m_gdb->ParticleBoxArray(lev);
     const auto& dm = m_gdb->DistributionMap(lev);
     BoxArray nba = ba;
-#if (!defined WARPX_DIM_RZ) || (!defined WARPX_USE_PSATD)
```

Comment:
This looks a bit intricate in the original source code, so maybe a good time to double-check it.

Reply:
@ax3l I noticed your suggestion to crosscheck this and I would like to reply. This line of code was introduced in #1240 in order to fix a bug with the PSATD solver in RZ geometry, related to the cell-centering of the diagnostics `rho` and `divE`.

Reply:
This check is needed so that the rho and divE are calculated with a consistent centering. With RZ spectral, everything is cell centered and the spectral transforms require cell centering (and divE is calculated by dividing by k in spectral space).

Reply:
Thank you both for your help!
I see, this is just an intricate way to write not(RZ && PSATD) :) I knew I rushed it, good that I marked it. Thanks again.